src/libraries/System.Collections/tests/Generic/Dictionary/Dictionary.Generic.Tests.cs

```diff
@@ -611,5 +613,37 @@ public void TrimExcess_Generic_DoesInvalidateEnumeration()
         }
 
         #endregion
+
+        #region Non-randomized comparers
+        [Fact]
+        public void Dictionary_Comparer_NonRandomizedStringComparers()
+        {
+            RunTest(null);
```

Comment:
would it be better to use `MemberData` in order to get separate pass/fail results for each of these cases?

Reply:
I tried that initially, but it was a bit cryptic to match the failure back to the argument instance. Maybe there's some secret sauce to getting a friendlier error message in that case?

Right now you get a line number as part of the exception if there's a failure. And while it's not ideal, at least it's a straightforward mapping.

Reply:
Fair enough. Possibly it would require some trick such as passing a 2nd parameter that was the comparer name.